### PR TITLE
lib: bsdlib: Add socket offloading for fcntl()

### PR DESCRIPTION
**Pull request by stig-bjorlykke**
Add function for fcntl() translation in socket offloading.

Signed-off-by: Stig Bjørlykke <stig.bjorlykke@nordicsemi.no>

### CODE REVIEW

**Comment by stig-bjorlykke** [comment]
It looks like ```nrf_fcntl()``` is defined in ```nrf_socket.h``` but not yet available in ```libbsd```. This PR has to wait for this to be solved.

File: lib/bsdlib/nrf91_sockets.c [hunk 1]
```diff
@@ -703,6 +704,42 @@ static int nrf91_socket_offload_getaddrinfo(const char *node,
...
+
+	default:
+		goto error;
+	}
+
+	return retval;
+
+error:
+	retval = -1;
+	errno = -ENOTSUP;
```
Comment (rlubos) [inline lib/bsdlib/nrf91_sockets.c]: `errno` value should not be negated. But I see it's an issue all across the file, we should fix this in a bulk in a separate PR.

And I'd consider `EINVAL` in this case, all in all that's arguemnts that are not correct. `ENOTSUP` is not specified by man pages as an error code for `fcntl`.

Comment (stig-bjorlykke) [reply lib/bsdlib/nrf91_sockets.c]: I followed the existing pattern for error handling in this file without thinking about the negative value returned.  I'll update this PR to return positive EINVAL in this case.

**Comment by rlubos** [comment]
Adding DNM so that it's easier to spot from the PR list that it's not ready for merge.

**Comment by rlubos** [comment]
Hey, an updated `bsdlib` version is now available for nrfconnect - please rebase and we should be good with the CI.

**Pull request merged**